### PR TITLE
:sparkles: Display Info if the Internal ProcessEngine is Unused

### PR DESCRIPTION
**Pull request by Paulomart**
## What did you change?

**Changes**:

- Display a info if the internal processengine is unused.
- Fix creating of main window.

PR: #664 

grafik

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/config-panel/config-panel.scss [hunk 1]
```diff
@@ -13,3 +13,13 @@
...
 }
+
+.use-internal-processengine {
+  cursor: pointer;
+  text-decoration: underline;
+}
+
+.config-panel__panel__alert {
+  margin-top: 10px;
+  margin-bottom: 0px;
```
Comment (LeoTT) [inline src/modules/config-panel/config-panel.scss]: `margin-bottom: 0`

File: src/modules/config-panel/config-panel.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import {EventAggregator, Subscription} from 'aurelia-event-aggregator';
 import {bindable, inject} from 'aurelia-framework';
+import {computedFrom} from 'aurelia-framework';
```
Comment (LeoTT) [inline src/modules/config-panel/config-panel.ts]: There's already an import from `aurelia-framework` one line above, you can move `computedFrom` there.

Comment (Vyperus) [inline src/modules/config-panel/config-panel.ts]: There is already an import from aurelia-framework. Move this import to the import above.

File: src/modules/config-panel/config-panel.ts [hunk 3]
```diff
@@ -80,4 +85,17 @@ export class ConfigPanel {
     this._router.navigateBack();
   }
 
+  @computedFrom('baseRoute')
+  public get isBaseRouteNotSetToInternalProcessEngine(): boolean {
```
Comment (LeoTT) [inline src/modules/config-panel/config-panel.ts]: I would remove the negation from the getter (even if the actual enduser of the getter is more interested in the negation).

If the negation is default, the positive becomes double negation. Also the name of the getter is complicated enough without the negation.

File: src/modules/config-panel/config-panel.ts [hunk 4]
```diff
@@ -80,4 +85,17 @@ export class ConfigPanel {
...
+    return this.internalProcessEngineBaseRoute !== this.baseRoute;
+  }
+
+  public hasInternalProcessEngineBaseRouteSet(): boolean {
+    return this.internalProcessEngineBaseRoute !== null;
+  }
+
+  public async setBaseRouteToInternalProcessEngine(): Promise<void> {
```
Comment (LeoTT) [inline src/modules/config-panel/config-panel.ts]: Why is this `async`?

File: src/modules/config-panel/config-panel.ts [hunk 5]
```diff
@@ -22,18 +23,22 @@ export class ConfigPanel {
   public config: typeof environment = environment;
   public isLoggedInToProcessEngine: boolean;
   @bindable() public baseRoute: string;
+  public internalProcessEngineBaseRoute: string | null;
```
Comment (LeoTT) [inline src/modules/config-panel/config-panel.ts]: 👍  for `| null`

**Pull request merged**